tidy3d/plugins/adjoint/components/medium.py

```diff
@@ -273,40 +326,76 @@ def store_vjp(
-            field_name = f"E{dim}"
 
             # grab the original data and its coordinatess
             orig_data_array = self.eps_dataset.field_components[eps_field_name]
             coords = orig_data_array.coords
 
+            grid = grids[eps_field_name]
+            d_sizes = grid.sizes
+            d_sizes = [d_sizes.x, d_sizes.y, d_sizes.z]
+
             # construct the coordinates for interpolation and selection within the custom medium
-            interp_coords = {dim_pt: coords[dim_pt] for dim_pt in "xyz" if len(coords[dim_pt]) > 1}
-            isel_coords = {dim_pt: 0 for dim_pt in "xyz" if len(coords[dim_pt]) <= 1}
+            # TODO: extend this to all points within the volume.
+            interp_coords = {}
+            sum_axes = []
 
-            # interpolate into the forward and adjoint fields along this dimension and dot them
-            e_fwd = grad_data_fwd.field_components[field_name]
-            e_adj = grad_data_adj.field_components[field_name]
-            e_dotted = (e_fwd * e_adj).isel(f=0, **isel_coords).interp(**interp_coords)
+            for dim_index, dim_pt in enumerate("xyz"):
 
-            # compute the size of the user-supplied medium along each dimension.
-            grid = grids[eps_field_name]
-            d_sizes = grid.sizes
+                coord_dim = coords[dim_pt]
 
-            # if any of the sizes are just 0, indicating an ndim < 3, just normalize out to 1.0
-            d_sizes = (
-                np.ones(1) if len(dl) == 1 and dl[0] <= 0 else dl
-                for dl in (d_sizes.x, d_sizes.y, d_sizes.z)
-            )
+                # if it's uniform / single pixel along this dim
+                if len(np.array(coord_dim)) == 1:
+                    # discretize along this edge like a regular volume
+
+                    # compute the length of the pixel within the sim bounds
+                    r_min_coords, r_max_coords = grid.boundaries.to_list[dim_index]
+                    r_min_sim, r_max_sim = np.array(sim_bounds).T[dim_index]
+                    r_min = max(r_min_coords, r_min_sim)
+                    r_max = min(r_max_coords, r_max_sim)
+                    size = abs(r_max - r_min)
+
+                    # discretize according to PTS_PER_WVL
+                    num_cells_dim = int(size * PTS_PER_WVL_INTEGRATION / wvl_mat) + 1
+
+                    # compute the length element along the dim, handling case of sim.size=0
+                    d_len = size / num_cells_dim if size > 0 else 1.0
+                    d_sizes[dim_index] = np.array([d_len])
+
+                    # construct the interpolation coordinates along this dimension
+                    coords_interp = np.linspace(r_min + d_len / 2, r_max - d_len / 2, num_cells_dim)
+                    interp_coords[dim_pt] = coords_interp
+                    sum_axes.append(dim_pt)
+
+                # otherwise
+                else:
+                    # just evaluate at a point
```

Comment:
`just evaluate at the original data coords` ?

Reply:
yea this is probably more clear.